metabase/metabase
# isomorphic viz settings computation

### Description

In the initial release of the migration to ECharts for isomorphic charts, we reused most of the code between client-side and GraalJS SSR. However, we still partially duplicated the visualization settings computation for static visualizations. This was because importing getComputedSettingsForSeries introduced frontend code that relied on browser APIs, causing runtime failures.

This PR enables static visualizations to fully reuse the settings computation logic, eliminating potential inconsistencies between client-side and static charts related to visualization settings. It also improves developer ergonomics by removing the need to write separate settings computations for static visualizations.

The same principle applies to the formatValue function, which we can now use directly in static visualizations instead of having a duplicated formatStaticValue. This eliminates the need to include formatValue in the rendering context.

### How to verify

- All visual Loki tests should be green
- BE specs that verify static-viz rendering with real data in GraalVM should be green
- Verify manually static viz rendering in dashboard subscriptions & question alerts works

### Checklist
